File: 2006-find-the-student-that-will-replace-the-chalk/find-the-student-that-will-replace-the-chalk.c
int chalkReplacer(int* chalk, int chalkSize, int k) {
    long long sum = 0;
    for (int i = 0; i < chalkSize; i++) {
        sum += chalk[i];
    }
    
    k = k % sum;  // Reduce k to be within one complete cycle
    
    for (int i = 0; i < chalkSize; i++) {
        if (k < chalk[i]) {
            return i;
        }
        k -= chalk[i];
    }
    return 0;
}

File: 2006-find-the-student-that-will-replace-the-chalk/find-the-student-that-will-replace-the-chalk.java
public class Solution {
    public int chalkReplacer(int[] chalk, int k) {
        long sum = 0;
        for (int chalkPiece : chalk) {
            sum += chalkPiece;
        }

        k = (int) (k % sum); // Reduce k to be within one complete cycle

        for (int i = 0; i < chalk.length; i++) {
            if (k < chalk[i]) {
                return i;
            }
            k -= chalk[i];
        }

        return 0;
    }
}